### DISCUSSION TITLE
Execute command line validations

### DISCUSSION THREAD

**Discussion by bdovaz** [dotnet/ResXResourceManager#606] [Q&A] [answered]
I see that there are a number of validations that are represented in the UI:
* Missing translations
* Inconsistent punctuation marks
* ...

I would like to be able to run these validations by command line to be able to introduce it in the CI process (PR, nightlies, ...).

Is there any way?

**Comment by tom-englert** [comment]
https://github.com/dotnet/ResXResourceManager/blob/master/Documentation/Topics/Scripting.md

**Comment by bdovaz** [comment]
@tom-englert but I see that the script that comes with it only works with Excel files? I want it to scan the resx files I have in the repository for errors based on the rules it has defined.

**Comment by tom-englert** [comment]
No, you can script anything

**Comment by bdovaz** [comment]
Can you give me an example or at least what classes do I need?

Thanks

**Comment by tom-englert** [comment]
It's just exposing the model, e.g.
https://github.com/dotnet/ResXResourceManager/blob/de9161d87ae0ee97bc9903a18a375a0d2bb1dcd1/src/ResXManager.Scripting/Sample.ps1#L23-L26
will get you here:
https://github.com/dotnet/ResXResourceManager/blob/de9161d87ae0ee97bc9903a18a375a0d2bb1dcd1/src/ResXManager.Model/ResourceManager.cs#L82-L85
so you can iterate over all items, checking the information you want

**Comment by bdovaz** [comment]
Thanks but I mean how can I import the resx files in order to get that table entries

In the sample ps1 there is only a importExcel reference

**Comment by tom-englert** [comment, marked as answer]
https://github.com/dotnet/ResXResourceManager/blob/de9161d87ae0ee97bc9903a18a375a0d2bb1dcd1/src/ResXManager.Scripting/Sample.ps1#L7-L13
